packages/medusa/src/services/product-variant-inventory.ts

```diff
@@ -638,28 +638,52 @@ class ProductVariantInventoryService extends TransactionBaseService {
 
   async setVariantAvailability(
     variants: ProductVariant[] | PricedVariant[],
-    salesChannelId: string | string[] | undefined
+    salesChannelId: string | string[] | undefined,
+    variantInventoryMap?: Map<string, ProductVariantInventoryItem[]>
   ): Promise<ProductVariant[] | PricedVariant[]> {
     if (!this.inventoryService_) {
       return variants
     }
 
+    if (!variantInventoryMap) {
```

Comment:
todo: replace by checking if there is a value in the map (!variantInventoryMap.size)